src/main/java/com/iexec/core/task/update/TaskUpdateRequestManager.java

```diff
@@ -70,35 +69,32 @@ public TaskUpdateRequestManager(TaskService taskService,
 
     /**
      * Publish TaskUpdateRequest async
+     * As of now, we do sequential requests to the DB which can cause a big load.
+     * We should aim to have some batch requests to unload the scheduler.
+     *
      * @param chainTaskId
      * @return
      */
-    public CompletableFuture<Boolean> publishRequest(String chainTaskId) {
-        Supplier<Boolean> publishRequest = () -> {
-            if (chainTaskId.isEmpty()){
-                return false;
-            }
-            if (queue.containsTask(chainTaskId)){
-                log.debug("Request already published [chainTaskId:{}]", chainTaskId);
-                return false;
-            }
-            final Optional<Task> oTask = taskService.getTaskByChainTaskId(chainTaskId);
-            if (oTask.isEmpty()) {
-                log.warn("No such task. [chainTaskId: {}]", chainTaskId);
-                return false;
-            }
+    public synchronized boolean publishRequest(String chainTaskId) {
+        if (chainTaskId.isEmpty()) {
+            return false;
+        }
+        if (queue.containsTask(chainTaskId)) {
+            log.debug("Request already published [chainTaskId:{}]", chainTaskId);
+            return false;
+        }
+        final Optional<Task> oTask = taskService.getTaskByChainTaskId(chainTaskId);
+        if (oTask.isEmpty()) {
+            log.warn("No such task. [chainTaskId: {}]", chainTaskId);
+            return false;
+        }
 
-            final Task task = oTask.get();
-            taskUpdateExecutor.execute(new TaskUpdate(task, this::updateTask));
-            log.debug("Published task update request" +
-                    " [chainTaskId:{}, currentStatus:{}, contributionDeadline:{}, queueSize:{}]",
-                    chainTaskId, task.getChainTaskId(), task.getContributionDeadline(), queue.size());
-            return true;
-        };
-        // TODO: find a better way to publish request.
-        // As of now, we do sequential requests to the DB which can cause a big load.
-        // We should aim to have some batch requests to unload the scheduler.
-        return CompletableFuture.supplyAsync(publishRequest, executorService);
+        final Task task = oTask.get();
+        taskUpdateExecutor.execute(new TaskUpdate(task, this::updateTask));
+        log.debug("Published task update request" +
+                        " [chainTaskId:{}, currentStatus:{}, contributionDeadline:{}, queueSize:{}]",
+                chainTaskId, task.getChainTaskId(), task.getContributionDeadline(), queue.size());
```

Comment:
It must have been here for quite some time
```suggestion
                chainTaskId, task.getCurrentStatus(), task.getContributionDeadline(), queue.size());
```
Would it be possible to complete the javadoc ?

Reply:
Please see https://github.com/iExecBlockchainComputing/iexec-core/pull/633/commits/8e619493c5683ed565c1feb8d86479b881bf9ac7.